File: java/src/main/java/com/openhealthhub/plandefinition/PlanDefinitionClient.java
package com.openhealthhub.plandefinition;

import ca.uhn.fhir.rest.client.api.IGenericClient;
import com.openhealthhub.util.FhirUtil;
import org.hl7.fhir.r4.model.Bundle;
import org.hl7.fhir.r4.model.PlanDefinition;

public class PlanDefinitionClient {

    private final IGenericClient client;

    public static void main(String... args) {
        PlanDefinitionClient client = new PlanDefinitionClient();

        String action = args.length == 0 ? "get" : args[1];

        if ("search".equals(action)) {
            FhirUtil.printResource(client.searchPlanDefinition());
            return;
        }

        FhirUtil.printResource(client.getPlanDefinition());
    }

    PlanDefinitionClient() {
        client = FhirUtil.createClient();
    }

    PlanDefinition getPlanDefinition() {
        return client.read()
                .resource(PlanDefinition.class)
                .withId("4944e73f-e447-49ba-a64c-a246b9ef4bdd")
                .execute();
    }

    public Bundle searchPlanDefinition() {
        return client.search()
                .forResource(PlanDefinition.class)
                .where(PlanDefinition.DEFINITION.hasId("Questionnaire/866683f3-c41b-47c0-b42f-86f9ff978d1d"))
                .and(PlanDefinition.PUBLISHER.matches().value("Program Creator"))
                .returnBundle(Bundle.class)
                .execute();
    }

}

File: php/src/OpenHealthhub/PhpFhirClient/PlanDefinitionClient.php
<?php


namespace OpenHealthhub\PhpFhirClient;


use DCarbone\PHPFHIRGenerated\R4\FHIRResource\FHIRBundle;
use DCarbone\PHPFHIRGenerated\R4\FHIRResource\FHIRDomainResource\FHIRPlanDefinition;

class PlanDefinitionClient
{
    public function getPlanDefinition($id): FHIRPlanDefinition
    {
        $client = new FhirClient();
        $res = $client->get(sprintf('PlanDefinition/%s', $id));
        return new FHIRPlanDefinition($res);
    }

    public function searchPlanDefinition($questionnaire, $publisher): FHIRBundle
    {
        $client = new FhirClient();
        $res = $client->search(sprintf('PlanDefinition?definition=Questionnaire/%s&publisher=%s', $questionnaire, $publisher));
        return new FHIRBundle($res);
    }
}
